### PR TITLE
Improve security and allow using different tabs

### PR DESCRIPTION
**Pull request by Cito**
- Check that the user account is still active where this is relevant
- Validate the dataset ID as an accession number
- Do not check the CSRF token on login (thereby allowing multiple browser tabs)

### CODE REVIEW

File: tests/integration/auth_adapter/test_sessions.py [hunk 1]
```diff
@@ -437,22 +437,34 @@ async def test_login_with_cookie_and_registered_data_steward(bare_client: BareCl
...
+    """Test login request with session cookie but no CSRF token.
+
+    It should be possible to login from a new browser tab using the session cookie.
+    """
+    setup_daos()
+
     store = await get_session_store(config=CONFIG)
-    session = await store.create_session(
-        ext_id="john@aai.org", user_name="John Doe", user_email="john@home.org"
+    session_dict = await store.create_session(
```
Comment (mephenor) [inline tests/integration/auth_adapter/test_sessions.py:448]: Not sure why this is called `session_dict` now, got confused for a second with the attribute access below

**Review by mephenor — APPROVED** [review]
Logic seems fine, just two comments on documentation and naming that was a bit confusing.

File: tests/integration/auth_adapter/test_sessions.py (hunk 1 above)
Comment (Cito) [reply tests/integration/auth_adapter/test_sessions.py]: I think I copied it from another test that had this misleading name. Fixed in 8bc136a.

File: src/auth_service/user_management/claims_repository/router.py [hunk 2]
```diff
@@ -310,7 +316,9 @@ async def grant_download_access(  # noqa: PLR0913
     """Grant download access permission for a dataset to a user with the given IVA.
 
     Note that at this point the IVA needs to exist, but does not need to be verified.
-    We also do not check here whether the dataset actually exists.
+    We check whether the user exists but we do not check whether the user is active.
```
[the remark this answers is not in the document]
Comment (mephenor) [reply src/auth_service/user_management/claims_repository/router.py:319]: This comment is a bit stale, I checked that this is indead dealt with by the IVA check.

[the remark this answers is not in the document]
Comment (Cito) [reply src/auth_service/user_management/claims_repository/router.py:319]: Renamed the method in b5303ea. Is that what you meant?

**Pull request merged**